notification-server/scripts/Dockerfile

```diff
@@ -0,0 +1,28 @@
+ARG build_image=rust:1.74-buster
+ARG base_image=debian:buster-slim
+FROM ${build_image} AS build
+
+WORKDIR /usr/app/notification-server
+
+COPY notification-server/Cargo.toml notification-server/Cargo.lock ./
+COPY deps /usr/app/deps
+RUN mkdir -p src/bin
+RUN echo 'fn main() { println!("Dummy!"); }' | tee ./src/bin/api.rs ./src/bin/service.rs
```

Comment:
This seems weird but is in reality quite effective to get faster build time, since the dependencies get cached and only gets reinstalled at changes instead of during at all source code changes. 

In another PR this method was led down, but I kind of like it, so will try again.